Dapper/DefaultTypeMap.cs

```diff
@@ -127,7 +127,7 @@ public ConstructorInfo FindExplicitConstructor()
         /// <returns>Mapping implementation</returns>
         public SqlMapper.IMemberMap GetConstructorParameter(ConstructorInfo constructor, string columnName)
         {
-            ParameterInfo param = MatchFirstOrDefault(constructor.GetParameters(), columnName, static p => p.Name);
+            ParameterInfo param = MatchFirstOrDefault(constructor.GetParameters(), columnName, static p => p.Name)!;

```

Comment:
This feels weird because if we are null we'll immediately throw downstream - should we have a null check with better messaging here?

Reply:
Yeah, I wondered about that. I was trying to avoid too many *functionally observable* changes, but it might be warranted here.